engine/src/test/java/io/zeebe/engine/processor/workflow/incident/MultiInstanceIncidentTest.java

```diff
@@ -115,9 +128,61 @@ public void shouldCreateIncidentIfInputVariableIsNotAnArray() {
         .hasElementId(elementInstance.getValue().getElementId())
         .hasErrorType(ErrorType.EXTRACT_VALUE_ERROR)
         .hasErrorMessage(
-            "Expected multi-instance input collection variable '"
+            "Expected result of the expression '"
                 + INPUT_COLLECTION
-                + "' to be an ARRAY, but found 'STRING'.");
+                + "' to be 'ARRAY', but was 'STRING'.");
+  }
+
+  @Test
+  public void shouldCreateIncidentIfOutputElementExpressionEvaluationFailed() {
+    // given
+    final long workflowInstanceKey =
+        ENGINE
+            .workflowInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariable(INPUT_COLLECTION, List.of(1, 2, 3))
+            .create();
+
+    // when
+    final List<Record<JobRecordValue>> created =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withType(jobType)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .asList();
+    assertThat(created).hasSize(3);
+
+    final Record<JobBatchRecordValue> jobs = ENGINE.jobs().withType(jobType).activate();
+    assertThat(jobs.getValue().getJobs()).hasSize(3);
+
+    for (final long key : jobs.getValue().getJobKeys()) {
+      ENGINE.job().withKey(key).complete();
+    }
+    final List<Record<JobRecordValue>> completed =
+        RecordingExporter.jobRecords(JobIntent.COMPLETED)
+            .withType(jobType)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .asList();
+    assertThat(completed).hasSize(3);
+
+    // then
+    final Record<IncidentRecordValue> incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .getFirst();
+
+    final Record<WorkflowInstanceRecordValue> elementInstance =
+        RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_COMPLETING)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .withElementId(ELEMENT_ID)
+            .getFirst();
+
+    Assertions.assertThat(incident.getValue())
+        .hasElementInstanceKey(elementInstance.getKey())
+        .hasElementId(elementInstance.getValue().getElementId())
+        .hasErrorType(ErrorType.EXTRACT_VALUE_ERROR)
+        .hasErrorMessage(
+            "failed to parse expression 'some_result': [1.12] error: "
+                + "'in' expected but end of source found\n\nsome_result\n           ^");
```

Comment:
This is not the failure you're looking for :robot: 

It seems that the FEEL engine doesn't like variable names starting with `some` (which is a reserved word in FEEL). 

Since the output element variable would be created when the multi-instance element is activated, we need to use some kind of expression (e.g. `sum(r)`).